### ISSUE TITLE
Add `StorageSpec` to `FunDef`

### ISSUE THREAD

**Issue by ivanperez-keera** [fdedden/language-c99-simple#22]
Unlike `FunDecln`, which is defined as:
```
FunDecl (Maybe StorageSpec) Type Ident [Param]
```

a `FunDef` does not include a `StorageSpec` (optional or not):
```
FunDef Type Ident [Param] [Decln] [Stmt]
```

This prevents us from adding the keyword `static` to local functions, which is a requirement for compliance with MISRA C 2012 in Copilot (see https://github.com/Copilot-Language/copilot/issues/472), which we really want to get.

**Comment by ivanperez-keera** [comment]
I think the changelog should have mentioned #22 (unless your changelogs mention PRs not issues) and the merge should have closed #22 (unless you don't close issues automatically on merge).

**Comment by fdedden** [comment]
Changelog should mention PR's, not issues as the issue does not change anything :).
